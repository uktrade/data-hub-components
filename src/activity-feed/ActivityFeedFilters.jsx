import React from 'react'
import PropTypes from 'prop-types'

import BasicActivityTypeFilter from './filters/BasicActivityTypeFilter'
import ShowDetailsFilter from './filters/ShowDetailsFilter'

export default class ActivityFeedFilters extends React.PureComponent {
  static propTypes = {
    isTypeFilterEnabled: PropTypes.bool.isRequired,
  }

  render() {
    const { isTypeFilterEnabled } = this.props

    if (isTypeFilterEnabled) {
      return <BasicActivityTypeFilter {...this.props} />
    } else {
      return <ShowDetailsFilter {...this.props} />
    }
  }
}
