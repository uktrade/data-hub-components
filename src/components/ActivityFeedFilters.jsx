import React from 'react'

export default class ActivityFeedFilters extends React.Component {
	render() {
    return (
      <form>
        <div>Show me <label><input type="checkbox" />My filters</label></div>
				<div>Activity type <select><option>All activity</option></select></div>
      </form>
    )
  }
}
