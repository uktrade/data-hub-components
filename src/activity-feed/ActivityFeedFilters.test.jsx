import React from 'react'
import renderer from 'react-test-renderer'

import ActivityFeedFilters from './ActivityFeedFilters'

describe('ActivityFeedFilters', () => {
  test('renders filters', () => {
    const tree = renderer
      .create(<ActivityFeedFilters />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
