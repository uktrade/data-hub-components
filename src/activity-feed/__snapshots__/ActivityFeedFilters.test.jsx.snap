// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityFeedFilters renders filters 1`] = `
<details
  className="sc-kEYyzF ihcnje"
  open={false}
  style={
    Object {
      "fontSize": "100%",
      "marginBottom": 0,
    }
  }
>
  <summary
    className="sc-kkGfuU cCdZFh"
  >
    <span
      className="sc-iAyFgw jRkIxn"
    >
      Filter and sort activity
    </span>
  </summary>
  <div
    className="sc-hSdWYo dfimje"
  >
    <form>
      <fieldset
        className="sc-gipzik eBgapt"
      >
        <legend
          className="sc-iRbamj eOWqGG"
        >
          Show me
        </legend>
        <div
          className="sc-hzDkRC hNOsPM"
        >
          <div
            className="sc-bRBYWo cfvpGy"
          >
            <div
              className="sc-Rmtcm fMfOqA"
            >
              <label
                className="sc-cSHVUG jqlwpQ"
              >
                <input
                  className="sc-kAzzGY gsGxQW"
                  type="checkbox"
                />
                <span
                  className="sc-chPdSV hLZCFS"
                >
                  My activity
                </span>
              </label>
              <label
                className="sc-cSHVUG jqlwpQ"
              >
                <input
                  className="sc-kAzzGY gsGxQW"
                  type="checkbox"
                />
                <span
                  className="sc-chPdSV hLZCFS"
                >
                  My team's activity
                </span>
              </label>
            </div>
          </div>
        </div>
      </fieldset>
    </form>
  </div>
</details>
`;
